modal_proto/api.proto

```diff
@@ -122,6 +122,8 @@ enum FunctionCallInvocationType {
   FUNCTION_CALL_INVOCATION_TYPE_UNSPECIFIED = 0;
   FUNCTION_CALL_INVOCATION_TYPE_SYNC_LEGACY = 1;
   FUNCTION_CALL_INVOCATION_TYPE_ASYNC_LEGACY = 2;
+  FUNCTION_CALL_INVOCATION_TYPE_SYNC = 3;
```

Comment:
The SYNC type isn't used in the PR at all. Figured it would be easier to track if both flags were added to the same client release.